File: src/BaseHmiTypes/Screens/Shapes/HmiEllipticalShapeBase.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public abstract class HmiEllipticalShapeBase : HmiCentricShapeBase
{
    public HmiProperty<double> RadiusX { get; set; } = 0;

    public HmiProperty<double> RadiusY { get; set; } = 0;
}

File: typescript/base-hmi-types/src/screens/shapes/HmiEllipticalShapeBase.ts
import { HmiProperty, staticProperty } from "../base.js";
import { HmiCentricShapeBase } from "./HmiCentricShapeBase.js";

export abstract class HmiEllipticalShapeBase extends HmiCentricShapeBase {
  radiusX: HmiProperty<number> = staticProperty(0);
  radiusY: HmiProperty<number> = staticProperty(0);
}
